### PR TITLE
Zed backports

### PR DESCRIPTION
**Pull request by Alex-Welsh**
Cherry picks:

```
git cherry-pick -x 0024b8948c  # Add role to deploy cASO
git cherry-pick -x 3cc89dc9bc  # Temporarily set kombu_reconnect_delay to 0.1
git cherry-pick -x 391aa4677f  # Adding optional delay between l3 agent restarts
git cherry-pick -x ea9e2853d1  # Prevent libvirtd reload when only generating config
git cherry-pick -x fd6e442670  # Changes for cluster-api magnum driver
git cherry-pick -x 1b1843d1ac  # Update ansible/roles/magnum/templates/magnum.conf.j2
git cherry-pick -x b7f9bd73b6  # config group has changed upstream
git cherry-pick -x eef3ff3084  # Add prometheus friendly hostnames
git cherry-pick -x 38337d4aa3  # remove whitespace
```

### CODE REVIEW

File: ansible/roles/caso/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    user: "{{ caso_keystone_user }}"
+    password: "{{ caso_keystone_password }}"
+    role: "admin"
+  {% endfor %}
+caso_ks_users: "{{ caso_ks_users_tmpl | from_yaml if caso_projects else [] }}"
+
+####################
+# Docker
+####################
+caso_install_type: "{{ kolla_install_type }}"
```
Comment (markgoddard) [inline ansible/roles/caso/defaults/main.yml:44]: No longer required.

File: ansible/roles/caso/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+caso_image: "{{ docker_registry ~ '/' if docker_registry else '' }}{{ docker_namespace }}/{{ kolla_base_distro }}-{{ caso_install_type }}-caso"
```
Comment (markgoddard) [inline ansible/roles/caso/defaults/main.yml:45]: Image name format has changed - see other roles.

File: ansible/roles/caso/templates/caso.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+projects = {{ caso_projects|join(', ') }}
+debug = {{ caso_logging_debug }}
+log_file = caso.log
+log_dir = {{ caso_log_dir }}
+log_rotation_type = none
+spooldir = /var/lib/caso
+
+[keystone_auth]
+auth_type = password
+auth_url = {{ admin_protocol }}://{{ kolla_internal_fqdn }}:{{ keystone_admin_port }}
```
Comment (markgoddard) [inline ansible/roles/caso/templates/caso.conf.j2:13]: The separate keystone admin port is deprecated in zed and gets removed in antelope. You can replace the whole thing with ``{{ keystone_internal_url }}`

File: ansible/roles/common/templates/conf/output/01-es.conf.j2 [hunk 4]
```diff
@@ -1,3 +1,17 @@
+{% if enable_caso | bool and inventory_hostname in groups['caso'] %}
+<match apel.events>
+    @type copy
+    <store>
+       @type elasticsearch
+       host { elasticsearch_address }}
+       port {{ elasticsearch_port }}
```
Comment (markgoddard) [inline ansible/roles/common/templates/conf/output/01-es.conf.j2:7]:
```suggestion
       @type opensearch
       host { opensearch_address }}
       port {{ opensearch_port }}
```

File: ansible/roles/magnum/tasks/config.yml [hunk 5]
```diff
@@ -67,6 +67,31 @@
   with_dict: "{{ magnum_services }}"
   notify:
     - Restart {{ item.key }} container
+    
+- name: Check if management cluster kubeconfig is provided
+  stat:
+    path: "{{ node_custom_config }}/magnum/kubeconfig"
+  register: kubeconfig
```
Comment (markgoddard) [inline ansible/roles/magnum/tasks/config.yml:74]:
```suggestion
  register: kubeconfig
  delegate_to: localhost
  run_once: true
```

File: ansible/roles/neutron/handlers/main.yml [hunk 6]
```diff
@@ -104,6 +150,11 @@
     healthcheck: "{{ service.healthcheck | default(omit) }}"
   when:
     - kolla_action != "config"
+    - inventory_hostname == item
+  loop: "{{ groups['neutron_l3_agent_running_True'] | default([]) }}"
```
Comment (markgoddard) [inline ansible/roles/neutron/handlers/main.yml:154]: Unfortunately this pattern no longer works with the version of Ansible used in Antelope. We can keep it in Zed though.

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: ansible/roles/neutron/handlers/main.yml (hunk 6 above)
Comment (Alex-Welsh) [reply ansible/roles/neutron/handlers/main.yml:154]: That's a shame. One to look out for in the future then.

File: ansible/roles/common/templates/conf/output/01-es.conf.j2 (hunk 4 above)
Comment (Alex-Welsh) [reply ansible/roles/common/templates/conf/output/01-es.conf.j2:7]: Fixed. I also replaced the other elasticsearch references in this file with opensearch, assuming they would follow the same format.

File: ansible/roles/caso/templates/caso.conf.j2 (hunk 3 above)
Comment (Alex-Welsh) [reply ansible/roles/caso/templates/caso.conf.j2:13]: Fixed

File: ansible/roles/caso/defaults/main.yml (hunk 2 above)
Comment (Alex-Welsh) [reply ansible/roles/caso/defaults/main.yml:45]: Fixed

File: ansible/roles/common/templates/conf/output/01-es.conf.j2 [hunk 7]
```diff
@@ -1,32 +1,46 @@
...
+       logstash_prefix apel
+       flush_interval 15s
+    </store>
+</match>
+{% endif %}
+
 <match **>
     @type copy
     <store>
-       @type elasticsearch
```
Comment (markgoddard) [inline ansible/roles/common/templates/conf/output/01-es.conf.j2:4]: I don't think these changes are necessary - there's another file for OpenSearch - 03-opensearch.conf.j2.

It does make me think that the apel.events bit should be in 03-opensearch.conf.j2 though.

**Review by markgoddard — CHANGES_REQUESTED** [review]

**Review by markgoddard — APPROVED** [review]

**Pull request merged**